pkg/application/doguRestartUseCase.go

```diff
@@ -0,0 +1,99 @@
+package application
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type DoguRestartUseCase struct {
+	doguInstallationRepository doguInstallationRepository
+	blueprintSpecRepo          blueprintSpecRepository
+	restartRepository          domainservice.DoguRestartRepository
+}
+
+func NewDoguRestartUseCase(doguInstallationRepository doguInstallationRepository, blueprintSpecRepo blueprintSpecRepository, restartRepository doguRestartRepository) *DoguRestartUseCase {
+	return &DoguRestartUseCase{doguInstallationRepository: doguInstallationRepository, blueprintSpecRepo: blueprintSpecRepo, restartRepository: restartRepository}
+}
+
+func (useCase *DoguRestartUseCase) TriggerDoguRestarts(ctx context.Context, blueprintId string) error {
+	logger := log.FromContext(ctx).WithName("DoguRestartUseCase.TriggerDoguRestarts")
+	blueprintSpec, err := useCase.blueprintSpecRepo.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("could not get blueprint spec by id: %q", err)
+	}
+
+	logger.Info("searching for Dogus that need a restart...")
+	var dogusThatNeedARestart []common.SimpleDoguName
+	allDogusNeedARestart := checkForAllDoguRestart(blueprintSpec)
+
+	if allDogusNeedARestart {
+		logger.Info("restarting all installed Dogus...")
+		err := useCase.restartAllInstalledDogus(ctx, logger)
+		if err != nil {
+			logger.Error(err, "could not restart all installed Dogus")
```

Comment:
You don't have to log errors. The reconciler already does this. If extra information is needed just wrap and return them with `fmt.Errorf`